### PR TITLE
Split user profile condition

### PR DESCRIPTION
**Pull request by dmitriim**

### CODE REVIEW

**Comment by dmitriim** [comment]
closes #74

File: classes/tool_cohortmanager/condition/user_profile.php [hunk 1]
```diff
@@ -48,13 +48,13 @@ class user_profile extends condition_base {
     /**
      * A list of supported default fields.
      */
-    private const SUPPORTED_STANDARD_FIELDS = ['auth', 'firstname', 'lastname', 'username', 'email',  'idnumber',
+    protected const SUPPORTED_STANDARD_FIELDS = ['auth', 'firstname', 'lastname', 'username', 'email',  'idnumber',
```
Comment (cameron1729) [inline classes/tool_cohortmanager/condition/user_profile.php:51]: Is it possible to get rid off this and use some core API to get all the standard profile fields?

File: classes/tool_cohortmanager/condition/user_profile.php [hunk 2]
```diff
@@ -48,13 +48,13 @@ class user_profile extends condition_base {
...
         'city', 'country', 'institution', 'department'];
 
     /**
      * A list of supported custom profile fields.
      */
-    private const SUPPORTED_CUSTOM_FIELDS = ['text', 'menu'];
+    protected const SUPPORTED_CUSTOM_FIELDS = ['text', 'menu'];
```
Comment (cameron1729) [inline classes/tool_cohortmanager/condition/user_profile.php:57]: We can get rid of this I think? It's not used here and the new class handles it.

File: classes/tool_cohortmanager/condition/user_profile.php (hunk 1 above)
Comment (dmitriim) [reply classes/tool_cohortmanager/condition/user_profile.php:51]: Good catch. I can't remember why I didn't include all core fields. There was a reason for sure as I thought about it.

File: classes/tool_cohortmanager/condition/user_profile.php (hunk 2 above)
Comment (dmitriim) [reply classes/tool_cohortmanager/condition/user_profile.php:57]: fixing...

Comment (dmitriim) [reply classes/tool_cohortmanager/condition/user_profile.php]: fixed

File: classes/tool_cohortmanager/condition/user_profile.php (hunk 1 above)
Comment (dmitriim) [reply classes/tool_cohortmanager/condition/user_profile.php:51]: ah. The reason was core changed APIs in different versions and I didn't want to have multiple branches so decided to pick a list of fields that would likely be used for the search.

Comment (cameron1729) [reply classes/tool_cohortmanager/condition/user_profile.php:51]: Makes sense, better to have one codebase that works across multiple versions than maintain a split.

**Pull request merged**